keps/sig-node/4939-tls-in-grpc-probe/README.md

```diff
@@ -0,0 +1,883 @@
+
+I would like to add new configuration fields alongside `port` and `service` in the 
+[Probe GRPCAction](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Probe).
+
+They can be used to indicate whether or not TLS should be used.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+1. Adds more code to Kubelet and surface area to `Pod.Spec`
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+We should add a boolean field to the
+[GRPCAction](https://github.com/kubernetes/kubernetes/blob/f422a58398d457a88ac7b05afca85a0e6b805605/pkg/apis/core/types.go#L2665-L2676)
+named `useTLS` which is optional and would default to `false`.
+
+I have identified the relevant code changes once the configuration
+structure is hammered out.
+
+Currently, the gRPC probe uses `insecure.NewCredentials()` when
+[establishing the connection](https://github.com/kubernetes/kubernetes/blob/d9441212d3e11dc13198f9d4df273c3555ecad11/pkg/probe/grpc/grpc.go#L59).
+
+If configured to use TLS, that `DialOption` should be replaced with:
+
+```go
+tlsConfig := &tls.Config{
+    InsecureSkipVerify: true,
+}
+tlsCredentials := credentials.NewTLS(tlsConfig)
+
+opts := []grpc.DialOption{
+    // ...
+    grpc.WithTransportCredentials(tlsCredentials),
+    // ...
+}
+```
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[ ] I/we understand the owners of the involved components may require updates to
```

Comment:
you gotta check the box :)  It's like your bank's EULA - what are you going to do, not sign it?

Reply:
oops forgot to do that, updated now